File: mobile/lib/models/account.dart
/// CareerMail AI — Email Account Model.
library;

class EmailAccount {
  final int id;
  final String emailAddress;
  final String provider;
  final bool isActive;
  final DateTime? lastSyncAt;
  final DateTime createdAt;

  EmailAccount({
    required this.id,
    required this.emailAddress,
    required this.provider,
    required this.isActive,
    this.lastSyncAt,
    required this.createdAt,
  });

  factory EmailAccount.fromJson(Map<String, dynamic> json) {
    return EmailAccount(
      id: json['id'] as int? ?? 0,
      emailAddress: json['email_address'] as String? ?? '',
      provider: json['provider'] as String? ?? 'gmail',
      isActive: json['is_active'] as bool? ?? true,
      lastSyncAt: json['last_sync_at'] != null
          ? DateTime.tryParse(json['last_sync_at'].toString())
          : null,
      createdAt: json['created_at'] != null
          ? DateTime.tryParse(json['created_at'].toString()) ?? DateTime.now()
          : DateTime.now(),
    );
  }
}

File: backend/app/schemas/account.py
"""
CareerMail AI — EmailAccount API schemas.

Never expose tokens or encrypted data through API responses.
"""

from __future__ import annotations

from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field


class AccountResponse(BaseModel):
    """Public account info — NO tokens, NO secrets."""

    id: int
    email_address: str
    provider: str
    is_active: bool
    last_sync_at: Optional[datetime] = None
    message_count: int = 0
    created_at: datetime

    model_config = {"from_attributes": True}


class AccountListResponse(BaseModel):
    """Response for GET /accounts."""

    accounts: list[AccountResponse]
    total: int


class OAuthStartResponse(BaseModel):
    """Response for GET /auth/google/start."""

    authorization_url: str
    message: str = "Open this URL in your browser to connect your Gmail account."


class SyncResponse(BaseModel):
    """Response for POST /accounts/{id}/sync."""

    account_id: int
    email_address: str
    new_messages: int = 0
    skipped_duplicates: int = 0
    total_messages_in_db: int = 0
    sync_duration_seconds: float = 0.0
    message: str = ""
